Add Lolin S2 Mini board

jhamhader opened the pull request:
Add ESP32-S2 Lolin/Wemos S2 Mini board.

Pins are taken from the official:
https://www.wemos.cc/en/latest/_static/files/sch_s2_mini_v1.0.0.pdf

# What does this implement/fix?

Add support ESP32-S2 Lolin/Wemos S2 Mini board.
Note, this depends on https://github.com/platformio/platform-espressif32/pull/736

Tested locally with a real Lolin/Wemos S2 Mini.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes [<link to issue>](https://github.com/esphome/feature-requests/issues/1626)

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** (TODO) esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [x] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
esp32:
  board: lolin_s2_mini
  type: esp-idf
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

jesserockz reviewed (approved):
Thanks

Noltari commented on esphome/components/esp32/boards.py:
```diff
@@ -523,6 +523,24 @@
     "lolin32_lite": {"LED": 22},
     "lolin_d32": {"LED": 5, "_VBAT": 35},
     "lolin_d32_pro": {"LED": 5, "_VBAT": 35},
+    "lolin_s2_mini": {
```
According to the schematic LED is on GPIO15, so it can also be added here...
```
"LED": 15
```

joegross replied:
@jhamhader Looks like the merge may be blocked on adding the above addition?

jhamhader replied:
Done

drapergeek commented:
Is there any momvement on this PR? I don't see any progress or issues.

derekyle commented:
Agreed, it would be nice to have this merged in. It looks like the pull request it was waiting on in the platformio repo has been merged in.

mkrsn commented:
@jhamhader thx for you Pull Request. I'm also waiting for this support. :dancers: 

It looks like the PR is blocked because there is the LED declaration missing in esphome/components/esp32/boards.py
Please see "Files Changed" comment from Noltari - line 526.

Would you please be so kind and add that? Thank you :)

ssieb commented:
This is probably still waiting for an upstream release and for esphome to move to that new version.

mkrsn commented:
@ssieb do you mean his PR https://github.com/platformio/platform-espressif32/pull/736 ?

This was already merged with v4.3.0 (latest: 5.1.1). Not sure what esphome currently use. But even if <v4.3.0, then it would be possible to go on without the recommended platform version and test at least.

LarsStegman commented:
@jesserockz Any chance this can get merged? I want to use these boards for bluetooth proxying HomeKit door contact sensors.

mkrsn commented:
@Noltari can you please confirm that the PR is only blocked because of the missing LED declaration?

In this case: The PR Owner seems to be unreachable or not willing to add the missing declaration. Is it ok if i redo the PR so that we get the new board on track?

Noltari commented:
> @Noltari can you please confirm that the PR is only blocked because of the missing LED declaration?
> 
> In this case: The PR Owner seems to be unreachable or not willing to add the missing declaration. Is it ok if i redo the PR so that we get the new board on track?

@mkrsn there has been a misunderstanding because I'm not blocking anything at all.
I just made a suggestion about adding the missing LED, but this can be merged and the LED can be added afterwards if @jhamhader can't be reached.

jhamhader commented:
I'll update the PR soon and we can merge :)

jhamhader committed:
Add Lolin S2 Mini board

Add ESP32-S2 Lolin/Wemos S2 Mini board.

Pins are taken from the official:
https://www.wemos.cc/en/latest/_static/files/sch_s2_mini_v1.0.0.pdf

jhamhader commented:
Rebased, fixed LED pin.

Not sure the default version includes https://github.com/platformio/platform-espressif32/pull/736

jhamhader commented:
@jesserockz can you please take a look?

Duckle29 commented:
I believe this is blocked because esphome uses V3.5 of platform-espressif32. s2 mini support wasn't added until 4.3, so my guess is this is waiting for esphome to move to at least that.

Any official word on if this is the hangup?
